docs/user_guide/expand_path.rst

```diff
@@ -71,6 +71,141 @@ If your session start time is present in your data path, you can indicate this f
 `1989 C standard format codes`_ for datetimes. For example, ``"{session_start_time:%Y-%m-%d}"`` will match
 ``"2021-01-02"`` and evaluate it to ``datetime.datetime(2021, 1, 2)``.
 
+Example Usage
+-----------------------
+
+Below are some full examples of how this feature can be used on some organizational patterns taken from real datasets.
+
+Example 1: `Allen Institute Visual Coding Dataset <https://registry.opendata.aws/allen-brain-observatory/>`_
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Allen Institute's Visual Coding dataset contains, among other data, motion-corrected videos of each
+experimental session, with the directory structure shown below.
+
+.. code-block:: bash
+
+    allen-brain-observatory/
+    ¦   visual-coding-2p/
+    ¦   +-- ophys_movies/
+    ¦   ¦   +-- ophys_experiment_645691416.h5
+    ¦   ¦   +-- ophys_experiment_<session_id>.h5
+    ¦   ¦   +-- ...
+
+To use :py:class:`~neuroconv.tools.path_expansion.LocalPathExpander` to find all ``ophys_movies`` files and
+extract their session IDs, you could use the following code.
+
+.. code-block:: python
+
+    >>> source_data_spec = {
+    ...     "allen-visual-coding": {
+    ...         "base_directory": "/allen-brain-observatory/visual-coding-2p",
+    ...         "file_path": "ophys_movies/ophys_experiment_{session_id}.h5"
+    ...     }
+    ... }
+    >>> metadata_list = path_expander.expand_paths(source_data_spec)
+    >>> for metadata in metadata_list:
+    ...     pprint(metadata)
+    {'metadata': {'NWBFile': {'session_id': '645691416'}},
+     'source_data': {'allen-visual-coding': {'file_path': '/allen-brain-observatory/visual-coding-2p/ophys_movies/ophys_experiment_645691416.h5'}}}
+    ...
+
+Example 2: `Buszaki Lab SenzaiY Dataset <https://app.globus.org/file-manager?origin_id=188a6110-96db-11eb-b7a9-f57b2d55370d&origin_path=%2FSenzaiY%2F>`_
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Buszaki Lab's SenzaiY dataset contains spiking and LFP data from mouse V1 with the directory structure
+shown below.
+
+.. code-block:: bash
+
+    SenzaiY/
+    ¦   YMV01/
+    ¦   +-- YMV01_170818/
+    ¦   ¦   +-- YMV01_170818.dat
```

Comment:
```suggestion
    ¦   ¦   +-- YMV01_170818.eeg
```

I think it's fine to limit the scope of this path expander to the spiking and LFP; the `.dat` files are actually the raw ecephys data and the `.eeg` files are the LFP though

Reply:
@CodyCBakerPhD Just making sure I understand this last bit - are you suggesting to replace the current demo using a `file_path` f-string to find `.dat` files with a demo using a `folder_path` f-string to find/parse the directories themselves instead? (since no additional info is in the file names)

Reply:
The `NeuroScopeRecordingInterface` uses a `file_path` convention (to a `.lfp` or `.eeg` file) whereas the `NeuroScopeSortingInterface` uses a `folder_path` convention

The demo code below this stops after running the `PathExpander` to get the resulting source data, but if you wanted to use that source data to initialize interfaces it would follow that pattern